packages/keycloak-user-management/src/components/UserGroupsList/index.tsx

```diff
@@ -0,0 +1,179 @@
+} from '../../ducks/userGroups';
+import { ADD_USER_GROUP, ERROR_OCCURED, USER_GROUPS_PAGE_HEADER, VIEW_DETAILS } from '../../lang';
+import { KEYCLOAK_URL_USER_GROUPS, SEARCH_QUERY_PARAM } from '../../constants';
+import {
+  fetchKeycloakUserGroups,
+  KeycloakUserGroup,
+  makeKeycloakUserGroupsSelector,
+} from '../../ducks/userGroups';
+
+/** Register reducer */
+reducerRegistry.register(keycloakUserGroupsReducerName, keycloakUserGroupsReducer);
+
+// Define selector instance
+const userGroupsSelector = makeKeycloakUserGroupsSelector();
+
+interface TableData {
+  key: number | string;
+  id: string | undefined;
+  name: string;
+}
+
+interface Props {
+  keycloakBaseURL: string;
+}
+
+/** default component props */
+const defaultProps = {
+  keycloakBaseURL: '',
+};
+
+/** Function which shows the list of all groups and their details
+ *
+ * @param {Object} props - UserGoupsList component props
+ * @returns {Function} returns User Groups list display
+ */
+export const UserGroupsList: React.FC<Props & RouteComponentProps> = (
+  props: Props & RouteComponentProps
+) => {
+  const dispatch = useDispatch();
+  const searchQuery = getQueryParams(props.location)[SEARCH_QUERY_PARAM] as string;
+  const getUserGroupsList = useSelector((state) =>
+    userGroupsSelector(state, { searchText: searchQuery })
+  );
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const { keycloakBaseURL } = props;
+
+  useEffect(() => {
+    if (isLoading) {
+      const serve = new KeycloakService(KEYCLOAK_URL_USER_GROUPS, keycloakBaseURL);
+      serve
+        .list()
+        .then((response: KeycloakUserGroup[]) => {
+          dispatch(fetchKeycloakUserGroups(response));
+        })
+        .catch(() => sendErrorNotification(ERROR_OCCURED))
+        .finally(() => setIsLoading(false));
+    }
+  });
+
+  if (isLoading)
+    return (
+      <Spin
+        size="large"
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          minHeight: '85vh',
+        }}
+      />
+    );
+
+  const searchFormProps = {
+    defaultValue: getQueryParams(props.location)[SEARCH_QUERY_PARAM],
+    onChangeHandler: createChangeHandler(SEARCH_QUERY_PARAM, props),
+  };
+
+  const tableData: TableData[] = getUserGroupsList.map(
+    (userGroup: KeycloakUserGroup, index: number) => {
+      return {
+        key: `${index}`,
+        id: userGroup.id,
+        name: userGroup.name,
+      };
+    }
+  );
+
+  const columns = [
+    {
+      title: 'Name',
+      dataIndex: 'name',
+      editable: true,
+      sorter: (a: TableData, b: TableData) => a.name.localeCompare(b.name),
+    },
+    {
+      title: 'Actions',
+      width: '10%',
+
+      // eslint-disable-next-line react/display-name
+      render: () => (
+        <span className="d-flex justify-content-end align-items-center">
+          <Link to="#">
+            <Button type="link" className="m-0 p-1">
+              Edit
```

Comment:
Let's move this to the language file

Reply:
`Edit` does not seem to be resolved for this, please confirm